### PR TITLE
CHANGELOG: update version recommendation

### PR DESCRIPTION
**Pull request by spzala**
Update version recommendation per the current policy.

For more details, see discussions under
https://github.com/etcd-io/website/pull/601

Signed-off-by: Sahdev Zala <spzala@us.ibm.com>

### CODE REVIEW

File: CHANGELOG/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Production recommendation
 
-The minimum recommended etcd versions to run in **production** are 3.3.18+, 3.4.2+, v3.5.3+. Refer to the [versioning policy](https://etcd.io/docs/v3.5/op-guide/versioning/) for more details.
+The minimum recommended etcd versions to run in **production** are 3.4.2+ and v3.5.3+. Refer to the [versioning policy](https://etcd.io/docs/v3.5/op-guide/versioning/) for more details.
```
Comment (ahrtr) [inline CHANGELOG/README.md:3]: There is a known data inconsistent issue (https://github.com/etcd-io/etcd/issues/11651) in both 3.4, and fxied in 3.4.8.  Although the data inconsistent issue (https://github.com/etcd-io/etcd/issues/13766) in 3.5 was fixed in 3.5.3, but there is a regression which is fixed in 3.5.4. 

So I would suggest to change the minimum recommended etcd version to 3.4.8+ and 3.5.4+.

Comment (spzala) [reply CHANGELOG/README.md:3]: @ahrtr ok, that sounds good. Thanks!

File: CHANGELOG/CHANGELOG-4.0.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 Previous change logs can be found at [CHANGELOG-3.x](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.x.md).
 
 
-The minimum recommended etcd versions to run in **production** are 3.2.28+, 3.3.18+, and 3.4.2+.
+The minimum recommended etcd versions to run in **production** are v3.4.8+ and v3.5.4+.
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-4.0.md:6]: I am thinking probably it makes more sense just to remove this in CHANGELOG-4.0, and keep the  recommendation in one place `CHANGELOG/README.md`

What do you think? @spzala

Comment (spzala) [reply CHANGELOG/CHANGELOG-4.0.md]: @ahrtr yes, good point! I am not sure why didn't we remove it when modified other files but probably it was missed. Thanks!

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @spzala

**Pull request merged**